tests/cmds/test_devices.py

```diff
@@ -0,0 +1,555 @@
+            "orgName": "Okta SSO",
+            "userExtRef": None,
+            "notes": None,
+            "active": True,
+            "blocked": False,
+            "emailPromo": True,
+            "invited": False,
+            "orgType": "ENTERPRISE",
+            "usernameIsAnEmail": True,
+            "creationDate": "2018-03-19T19:43:16.742Z",
+            "modificationDate": "2018-10-26T20:22:05.726Z",
+            "passwordReset": False,
+            "localAuthenticationOnly": False,
+            "licenses": ["admin.securityTools"],
+        },
+        {
+            "userId": 1014,
+            "userUid": "836473273124890369",
+            "status": "Active",
+            "username": "qatest@code42.com",
+            "email": "qatest@code42.com",
+            "firstName": "Chad",
+            "lastName": "Valentine",
+            "quotaInBytes": -1,
+            "orgId": 1017,
+            "orgUid": "836473214639515393",
+            "orgName": "Holy SaaS-a-roli",
+            "userExtRef": None,
+            "notes": None,
+            "active": True,
+            "blocked": False,
+            "emailPromo": True,
+            "invited": False,
+            "orgType": "ENTERPRISE",
+            "usernameIsAnEmail": True,
+            "creationDate": "2018-02-22T18:35:23.217Z",
+            "modificationDate": "2018-04-25T11:12:11.504Z",
+            "passwordReset": False,
+            "localAuthenticationOnly": False,
+            "licenses": ["admin.securityTools"],
+        },
+    ],
+}
+
+
+def _create_py42_response(mocker, text):
+    response = mocker.MagicMock(spec=Response)
+    response.text = text
+    response._content_consumed = mocker.MagicMock()
+    response.status_code = 200
+    return Py42Response(response)
+
+
+@pytest.fixture
+def mock_device_settings(mocker, mock_backup_set):
+    device_settings = mocker.MagicMock()
+    device_settings.name = "testname"
+    device_settings.guid = "1234"
+    device_settings.backup_sets = [mock_backup_set, mock_backup_set]
+    return device_settings
+
+
+@pytest.fixture
+def mock_backup_set(mocker):
+    backup_set = mocker.MagicMock()
+    backup_set["name"] = "test_name"
+    backup_set.destinations = {"destination_guid": "destination_name"}
+    backup_set.excluded_files = ["/excluded/path"]
+    backup_set.included_files = ["/included/path"]
+    backup_set.filename_exclusions = [".*\\.excluded_filetype"]
+    backup_set.locked = True
+    return backup_set
+
+
+@pytest.fixture
+def deactivate_response(mocker):
+    return _create_py42_response(mocker, "")
+
+
+@pytest.fixture
+def device_info_response(mocker):
+    return _create_py42_response(mocker, TEST_DEVICE_RESPONSE)
+
+
+@pytest.fixture
+def archives_list_generator(mocker):
+    yield TEST_ARCHIVES_RESPONSE
+
+
+@pytest.fixture
+def devices_list_generator(mocker):
+    return [TEST_COMPUTER_PAGE]
+
+
+@pytest.fixture
+def users_list_generator(mocker):
+    return [TEST_USERS_LIST_PAGE]
```

Comment:
These don't need to be fixtures, they can just be regular generator functions:

```
def devices_list_generator():
     yield TEST_COMPUTER_PAGE
```

Then just call it in the test:

```
cli_state.sdk.devices.get_all.return_value = devices_list_generator()
```

Otherwise the mock return value ends up just being a regular list, and if the real code does something like call `next()` on the generator these tests would break unnecessarily.

Reply:
updated